nabu-log-exporter/src/main/java/com/github/millerm/NabuLogExporter.java

```diff
@@ -0,0 +1,261 @@
+
+    public String getLogFilePath() {
+        return LOG_PATH_DIR.toString();
+    }
+
+    public String getStateFilePath() {
+        return stateFilePath.toString();
+    }
+
+    @Override
+    public void run() throws Exception {
+        LOG.info("Starting NabuLogExporter...");
+
+        while (true) {
+            processLogs();
+        }
+    }
+
+    @Override
+    public void exportLogs(String log) {
+        LOG.info("Exporting log: " + log);
+    }
+
+    @Override
+    public void processLogs() {
+        try (DirectoryStream<Path> stream = Files.newDirectoryStream(LOG_PATH_DIR)) {
+            LOG.info("Checking for existing files in log path: " + LOG_PATH_DIR);
+
+            for (Path entry : stream) {
+                if (Files.isRegularFile(entry)) {
+                    if (Files.size(entry) < MAX_LOG_FILE_SIZE) {
+                        executorService.submit(new LogTask(entry, stateFilePath));
+                    }
+
+                }
+            }
+        } catch (IOException e) {
+            LOG.severe("Error reading existing files: " + e.getMessage());
+        }
+
+        try (WatchService watchService = FileSystems.getDefault().newWatchService()) {
+            LOG_PATH_DIR.register(watchService, StandardWatchEventKinds.ENTRY_CREATE);
+
+            LOG.info("Watching for new log files at: " + LOG_PATH_DIR);
+
+            while (true) {
+                WatchKey key;
+                try {
+                    key = watchService.take();
+                } catch (InterruptedException e) {
+                    Thread.currentThread().interrupt();
+                    LOG.severe("Directory watching interrupted");
+                    return;
+                }
+
+                for (WatchEvent<?> event : key.pollEvents()) {
+                    WatchEvent.Kind<?> kind = event.kind();
+
+                    if (kind == StandardWatchEventKinds.OVERFLOW) {
+                        continue;
+                    }
+
+                    @SuppressWarnings("unchecked")
+                    WatchEvent<Path> ev = (WatchEvent<Path>) event;
+                    Path filename = ev.context();
+                    Path filePath = LOG_PATH_DIR.resolve(filename);
+
+                    if (kind == StandardWatchEventKinds.ENTRY_CREATE) {
+                        LOG.info("New file detected: " + filePath);
+                        executorService.submit(new LogTask(filePath, stateFilePath));
+                    }
+                }
+
+                boolean valid = key.reset();
+                if (!valid) {
+                    break;
+                }
+            }
+        } catch (IOException e) {
+            LOG.severe("Error watching directory: " + e.getMessage());
+        }
+    }
+
+    private static class LogTask implements Runnable {
+        private final Path filePath;
+        private final String stateFilePath;
+        long filePointer;
+
+        public LogTask(Path filePath, Path stateFilePath) {
```

Comment:
Should `stateFilePath` be called `stateDirPath`?

Reply:
I think that's a better name, great suggestion.